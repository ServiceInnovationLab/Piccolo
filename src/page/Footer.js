import React, { Fragment } from 'react';
import styled from 'styled-components';
import Container from '../elements/Container';
import List, { Item } from '../elements/List';
import { CreativeCommons } from '../components/CreativeCommons';

const Footer = styled.section`
  padding-top: 45px;
  height: 270px;
  background: #eee;
`;


const PageFooter = () => (
  <Fragment>
    <Footer>
      <Container>
        <List stripped bordered>
          <Item>About this website</Item>
          <Item>Contact Us</Item>
        </List>

        <CreativeCommons label="Content about creative commons"/>

      </Container>
    </Footer>
  </Fragment>
);

export default PageFooter;
