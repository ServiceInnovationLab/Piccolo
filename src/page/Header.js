import React, { Fragment } from 'react';
import styled from 'styled-components';
import Container from '../elements/Container';
import { H2 } from '../elements/Headings';
import Paragraph from '../elements/Paragraph';

const Header = styled.section`
  height: 135px;
  padding: 10px;
  background: #24273a;
  border-bottom: 12px solid #42e2ee;
  @media (min-width: 375px) {
    height: 95px;
  }
  @media (min-width: 768px) {
    height: 75px;
  }
`;

const Logo = styled.span`
  line-height: 45px;
  font-size: 34px;
  color: #fff;
  text-shadow: 0px 0px 20px #08fcff;
  font-weight: bold;
  @media (min-width: 768px) {
    line-height: 75px;
  }
`;

export const PageHeader = () => (
  <Fragment>
    <Header>
      <Container>
        <Logo>Legislation as Code - Demo</Logo>
      </Container>
    </Header>
  </Fragment>
);

export const SectionHeader = props => (
  <Fragment>
    <Container>
      <H2>{props.title}</H2>
      <Paragraph medium>{props.subtitle}</Paragraph>
      <hr />
    </Container>
  </Fragment>
);
