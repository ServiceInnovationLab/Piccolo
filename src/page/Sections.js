import React, { Fragment } from 'react';
import Paragraph from '../elements/Paragraph';
import Container from '../elements/Container';
import Section from '../elements/Section';
import { H2 } from '../elements/Headings';
import Grid, { Column } from '../elements/Grid';
import MadLib from '../components/Madlib';
import { IconCircle } from '../elements/Icon';
import List, { Item } from '../elements/List';
import countries from '../data/countries';

export const SectionOne = () => <Section light>
  <Container>
    <Paragraph small>
      Welcome to the Legislation as Code demonstrator.
    </Paragraph>
    <Paragraph small>
      There are many rules about how we do things around the world. Some of these are coded into law, such as voting ages, tax rates, and access to services. Sometimes these rules can be really complicated for a human to sit down and work out, but really easy for a machine to do.</Paragraph>

    <Paragraph small>
      Legislation as code is when we turn those rules into things machines can use so they can work things out for us.
    </Paragraph>

  </Container>
</Section>;

const EligibleHeader = props => <Paragraph medium adjust>
  You are eligible for a pension at age <span>{props.age}</span>
</Paragraph>;

export const SectionThree = props => <Fragment>
  {props.values.number_of_children && props.values.years_worked && props.values.years_lived_in_country && props.values.gender && <Fragment>
    <Section light>
      <Container>
        <MadLib values={props.values} />
      </Container>
    </Section>
    <SectionFour />
  </Fragment>}
</Fragment>;

export const SectionFour = props => <div style={{ display: props.show }}><Section dark>
  <Container>
    <Paragraph style={{ color: '#fff', textAlign: 'center' }}>
      Based on the information you entered, here is your eligibility for government pensions if you were applying in one of the following countries.
    </Paragraph>
  </Container>
</Section></div>;

export class SectionFive extends React.Component {
  render() {
    let date_range = '2018-08';

    return (
      <Section light>
        <Container>
          <Grid>
            {countries.map((country, i) => <Column key={i}>
              <H2>{country.name}</H2>
              <Eligibility
                country={country.label}
                age={this.props[country.label].data.persons.Tahi[country.age_key][date_range]}
                eligible={this.props[country.label].data.persons.Tahi[country.eligible_key][date_range]}
              />
              <Paragraph>{country.subtitle}</Paragraph>
              <List>
                {country.eligibility_list.map((item, i) => <Item key={i}>{item}</Item>)}
              </List>
            </Column>)}
          </Grid>
        </Container>
      </Section>
    );
  }
}

class Eligibility extends React.Component {
  render() {
    return <Fragment>
      {this.props.eligible && <Fragment>
        <EligibleHeader age={this.props.age} />
        <IconCircle value={this.props.age} />
      </Fragment>}

      {!this.props.eligible && <Fragment>
        <H2>You are not eligible for a pension</H2>
      </Fragment>}
    </Fragment>;
  }
}
