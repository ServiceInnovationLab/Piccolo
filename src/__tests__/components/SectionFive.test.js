import React from 'react';
import renderer from 'react-test-renderer';
import { SectionFive } from '../../page/Sections';
import 'jest-styled-components';

test('SectionFive renders correctly', () => {
  let israel = {
    data: {
      households: {
        one: {
          parents: [
            'Tahi'
          ]
        }
      },
      persons: {
        Tahi: {
          born_in_israel: {
            '2018-08': true
          },
          gender: {
            '2018-08': 'woman'
          },
          is_resident: {
            '2018-08': true
          },
          pension_contributing_years: {
            '2018-08': 66
          },
          pension_eligibility_age: {
            '2018-08': 62
          },
          eligible_for_pension: {
            '2018-08': true
          }
        }
      }
    },
    status: 200
  };
  let new_zealand = {
    data: {
      families: {
        family: {
          others: [
            'Tahi'
          ]
        }
      },
      persons: {
        Tahi: {
          is_permanent_resident: {
            '2018-08': true
          },
          super__eligibility: {
            '2018-08': true
          },
          super___eligibility_age: {
            '2018-08': 65
          },
          total_number_of_years_lived_in_nz_since_age_20: {
            '2018-08': 66
          },
          total_number_of_years_lived_in_nz_since_age_50: {
            '2018-08': 66
          }
        }
      },
      titled_properties: {
        house: {
          others: [
            'Tahi'
          ]
        }
      }
    },
    status: 200
  };
  let uruguay = {
    data: {
      households: {
        one: {
          parents: [
            'Tahi'
          ]
        }
      },
      persons: {
        Tahi: {
          edad_de_jubilacion: {
            '2018-08': 0
          },
          eligible_for_pension: {
            '2018-08': false
          },
          gender: {
            '2018-08': 'woman'
          },
          number_of_children: {
            '2018-08': 66
          },
          number_of_years_worked: {
            '2018-08': 66
          }
        }
      }
    },
    status: 200
  };

  const tree = renderer.create(<SectionFive
    israel={israel}
    new_zealand={new_zealand}
    uruguay={uruguay}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
