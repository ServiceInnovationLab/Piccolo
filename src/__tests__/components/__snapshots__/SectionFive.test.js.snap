// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionFive renders correctly 1`] = `
.c5 {
  font-size: 24px;
  color: #000;
  line-height: 1.5;
  font-size: 24px;
  height: 100%;
}

.c7 {
  font-size: 24px;
  color: #000;
  line-height: 1.5;
  height: auto;
}

.c1 {
  margin: 0 auto;
  width: 75%;
}

.c0 {
  font-family: sans-serif;
  padding: 55px 0;
  background: #d9f8fa;
}

.c0 p:first-child {
  margin-top: 0;
}

.c0 h2 {
  color: #2f3045;
  margin-bottom: 0;
}

.c0 h2 ~ p {
  margin-top: 10px;
  color: #000;
  padding-bottom: 35px;
}

.c0 p ~ hr {
  color: #000;
  height: 1px;
  border: 0;
  border-top: 1px solid #000;
}

.c0 h3 {
  color: #2f3045;
  font-weight: 400;
}

.c0 h2 ~ h3 {
  margin-top: 0;
}

.c4 {
  font-size: 34px;
}

.c2 {
  margin: 30px 0;
}

.c3 {
  padding: 0 40px 10px 40px;
  border-top: 1px solid #000;
  padding-top: 40px;
  margin-top: 50px;
}

.c3 h2:first-child {
  margin-top: 0;
}

.c3 h2 + p {
  margin-top: 20px;
}

.c3 h2 ~ p {
  margin-bottom: 0;
}

.c3 p:last-child {
  margin-bottom: 0;
  padding-bottom: 0;
}

.c6 {
  width: 100px;
  height: 100px;
  border-radius: 50%;
  background: #50a3d7;
  margin: 0 auto;
  margin-bottom: 40px;
}

.c6 > span {
  width: 100px;
  display: inline-block;
  font-size: 45px;
  font-weight: bold;
  color: #fff;
  text-align: center;
  line-height: 100px;
}

.c8 {
  padding-left: 25px;
}

.c9 {
  font-size: 22px;
  margin-bottom: 18px;
}

@media (min-width:1024px) {
  .c5 {
    height: 55px;
  }
}

@media (min-width:1024px) {
  .c7 {
    height: auto;
  }
}

@media (min-width:1024px) {
  .c1 {
    width: 88%;
  }
}

@media (min-width:1440px) {
  .c1 {
    width: 65%;
  }
}

@media (min-width:1024px) {
  .c0 {
    text-align: left;
  }
}

@media (min-width:1024px) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:1024px) {
  .c3 {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
    border-left: 1px solid #000;
    border-top: none;
    padding-top: 0;
  }
}

<section
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <h2
          className="c4"
        >
          New Zealand
        </h2>
        <p
          className="c5"
        >
          You are eligible for a pension at age 
          <span>
            65
          </span>
        </p>
        <div
          className="c6"
        >
          <span>
            65
          </span>
        </div>
        <p
          className="c7"
        >
          Government pension is called  New Zealand Superannuation, known as NZ Super for short.
        </p>
        <ul
          className="c8"
        >
          <li
            className="c9"
          >
            Gender does not determine eligibility in New Zealand.
          </li>
          <li
            className="c9"
          >
            You don't have to be retired from work to get NZ Super as it is not income tested.
          </li>
          <li
            className="c9"
          >
            Overseas benefits or pensions will generally be deducted from NZ Super.
          </li>
          <li
            className="c9"
          >
            To get NZ Super you must have lived in NZ for at least 10 years since you turned 20. Five of those years must be since you turned 50.
          </li>
        </ul>
      </div>
      <div
        className="c3"
      >
        <h2
          className="c4"
        >
          Israel
        </h2>
        <p
          className="c5"
        >
          You are eligible for a pension at age 
          <span>
            62
          </span>
        </p>
        <div
          className="c6"
        >
          <span>
            62
          </span>
        </div>
        <p
          className="c7"
        >
          Government pensions are known as old age pensions, sometimes referred to as insurance.
        </p>
        <ul
          className="c8"
        >
          <li
            className="c9"
          >
            Eligibility is determined by gender, payments to old age insurance, income test and the type of employment you have been in.
          </li>
          <li
            className="c9"
          >
            You must have been living in Israel for at least 5 years before you retire.
          </li>
          <li
            className="c9"
          >
            There are two stages of pensions; one at age 62 for women and 67 for men, and a second age pension at 70 which has less eligibility requirements than the first age pension.
          </li>
        </ul>
      </div>
      <div
        className="c3"
      >
        <h2
          className="c4"
        >
          Uruguay
        </h2>
        <h2
          className="c4"
        >
          You are not eligible for a pension
        </h2>
        <p
          className="c7"
        >
          Government pensions for work retirement are called Jubilaciones.
        </p>
        <ul
          className="c8"
        >
          <li
            className="c9"
          >
            Eligibility is determined by age and contributions to social security related to labour.
          </li>
          <li
            className="c9"
          >
            You must have completed 30 years of work.
          </li>
          <li
            className="c9"
          >
            There are two stages of pensions; one at age 60, and a second advanced age pension at 70.
          </li>
          <li
            className="c9"
          >
            For women each child (up to max of 5) counts as a 'year of work'.
          </li>
        </ul>
      </div>
    </div>
  </div>
</section>
`;
